cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -0,0 +1,157 @@
+
+namespace oneapi::dal::knn::backend {
+
+using dal::backend::context_gpu;
+namespace daal_classifier = daal::algorithms::classifier;
+
+template <typename Task>
+using descriptor_t = detail::descriptor_base<Task>;
+
+namespace daal_knn = daal::algorithms::bf_knn_classification;
+namespace interop = dal::backend::interop;
+
+template <typename Float>
+using daal_knn_brute_force_kernel_t =
+    daal_knn::prediction::internal::KNNClassificationPredictKernelUCAPI<Float>;
+
+template <typename Float, typename Task>
+static infer_result<Task> call_daal_kernel(const context_gpu& ctx,
+                                           const descriptor_t<Task>& desc,
+                                           const table& data,
+                                           const model<Task>& m) {
+    auto& queue = ctx.get_queue();
+    interop::execution_context_guard guard(queue);
+
+    const std::int64_t row_count = data.get_row_count();
+    const std::int64_t neighbor_count = desc.get_neighbor_count();
+
+    auto arr_responses = array<Float>{};
+    auto arr_indices = array<std::int64_t>{};
+    auto arr_distance = array<Float>{};
+
+    const auto daal_data = interop::convert_to_daal_table(queue, data);
+
+    auto daal_responses = daal::data_management::NumericTablePtr();
+    auto daal_indices = daal::data_management::NumericTablePtr();
+    auto daal_distance = daal::data_management::NumericTablePtr();
+
+    const auto data_use_in_model = daal_knn::doNotUse;
+    daal_knn::Parameter daal_parameter(
+        dal::detail::integral_cast<std::size_t>(desc.get_class_count()),
+        dal::detail::integral_cast<std::size_t>(desc.get_neighbor_count()),
+        data_use_in_model);
+
+    if (desc.get_result_options() & result_options::responses) {
+        if constexpr (std::is_same_v<Task, task::classification>) {
+            arr_responses = array<Float>::empty(queue, 1 * row_count, sycl::usm::alloc::device);
+            daal_responses = interop::convert_to_daal_table(queue, arr_responses, row_count, 1);
+        }
+    }
+    else {
+        daal_parameter.resultsToEvaluate = daal_classifier::none;
+    }
+
+    if (desc.get_result_options() & result_options::indices) {
+        daal_parameter.resultsToCompute |= daal_knn::computeIndicesOfNeighbors;
+        arr_indices =
+            array<std::int64_t>::empty(queue, neighbor_count * row_count, sycl::usm::alloc::device);
+        daal_indices =
+            interop::convert_to_daal_table(queue, arr_indices, row_count, neighbor_count);
+    }
+
+    if (desc.get_result_options() & result_options::distances) {
+        daal_parameter.resultsToCompute |= daal_knn::computeDistances;
+        arr_distance =
+            array<Float>::empty(queue, neighbor_count * row_count, sycl::usm::alloc::device);
+        daal_distance =
+            interop::convert_to_daal_table(queue, arr_distance, row_count, neighbor_count);
+    }
+
+    auto distance_impl = detail::get_distance_impl(desc);
```

Comment:
I think, need to check distance before creating tables

Reply:
Fixed